how-to-guides/e2e-tracking/notebooks/e2e_tracking.ipynb

```diff
@@ -0,0 +1,749 @@
+    "## Install Neptune and dependencies"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install -q -U neptune[sklearn,optuna] scikit-learn optuna evidently matplotlib\n",
+    "%pip install -q --user scikit-learn matplotlib"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run this if you get a `RuntimeError: main thread is not in main loop` error\n",
+    "import matplotlib\n",
+    "\n",
+    "matplotlib.use(\"Agg\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Track model training\n",
+    "\n",
+    "In this section, you will\n",
+    "1. Use Optuna to train multiple scikit-learn regression models\n",
+    "2. Leverage Neptune's [scikit-learn](https://docs.neptune.ai/integrations/sklearn/) and [Optuna](https://docs.neptune.ai/integrations/optuna/) integrations to automatically log metadata and metrics to Neptune for easy run comparison, while also using Neptune's [model registry](https://docs.neptune.ai/model_registry/overview/) to track models."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prepare the dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.datasets import fetch_california_housing\n",
+    "\n",
+    "data, target = fetch_california_housing(return_X_y=True)\n",
+    "X_train, X_test, y_train, y_test = train_test_split(data, target, test_size=0.25)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create or initialize a Neptune project\n",
+    "This can also be done from the UI 👉 https://docs.neptune.ai/setup/creating_project/\n",
+    "\n",
+    "This step also assumes that your [Neptune API token is set as an environment variable](https://docs.neptune.ai/setup/setting_api_token/).\n",
+    "\n",
+    "**Note:** The \"common/e2e-tracking\" project used here is read-only. To log to your own project, just update `project_name`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import neptune\n",
+    "from neptune.exceptions import ProjectNotFoundWithSuggestions\n",
+    "\n",
+    "project_name = \"common/e2e-tracking\"\n",
+    "%env NEPTUNE_PROJECT=$project_name\n",
+    "\n",
+    "try:\n",
+    "    # Initialize the project if it already exists\n",
+    "    project = neptune.init_project()\n",
+    "except ProjectNotFoundWithSuggestions:\n",
+    "    # Create the project if it does not already exist\n",
+    "    from neptune import management\n",
+    "\n",
+    "    management.create_project(name=project_name)\n",
+    "    project = neptune.init_project()"
```

Comment:
Are we sure this is the best approach? I'm hesitant to incorporate an exception into a usage example, and this just does not seem like the best way to create a new project.

I'd rather we assume a project exists, but we can still keep the `common` project as an alternative.